library/std/src/primitive_docs.rs

```diff
@@ -445,7 +445,26 @@ mod prim_unit {}
 /// Note that here the call to [`drop`] is for clarity - it indicates
 /// that we are done with the given value and it should be destroyed.
 ///
-/// ## 3. Get it from C.
+/// ## 3. Create it using `ptr::addr_of!`
+///
+/// Instead of coercing a reference to a raw pointer, you can use the macros
+/// [`ptr::addr_of!`] (for `*const T`) and [`ptr::addr_of_mut!`] (for `*mut T`).
+/// Those macros have an advantage over the coercion: you can create raw
+/// pointers to unaligned fields, which might be necessary if packed structs
+/// are involved.
```

Comment:
How about something like this?
```suggestion
/// Instead of coercing a reference to a raw pointer, you can use the macros
/// [`ptr::addr_of!`] (for `*const T`) and [`ptr::addr_of_mut!`] (for `*mut T`).
/// These macros allow you to create raw pointers to fields to which you cannot
/// create a reference (without causing undefined behaviour), such as an unaligned field.
/// This might be necessary if packed structs or uninitialized memory is involved.
```

Reply:
Included in latest (squashed) commit. I think nearly all of the final text comes from a reviewer suggestion 😉